tests/unit/test_option_parser.py

```diff
@@ -0,0 +1,187 @@
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+""" Test the option parsing and filtering form the compilation commands. """
+
+import unittest
+
+from codechecker_lib import option_parser
+from codechecker_lib.option_parser import ActionType
+
+
+class OptionParserTest(unittest.TestCase):
+    """
+    Test the option parser module which handles the
+    parsing of the g++/gcc compiler options.
+    """
+
+    def test_build_onefile(self):
+        """
+        Test the build command of a simple file.
+        """
+        source_files = ["main.cpp"]
+        build_cmd = "g++ -o main " + ' '.join(source_files)
+
+        res = option_parser.parse_options(build_cmd)
+        print(res)
+        self.assertTrue(len(source_files), len(res.files))
+        self.assertTrue(set(source_files) & set(res.files))
```

Comment:
These two assertions are true if the two sets have the same same size and they have at least one common element. Is there any reason why we don't check equality between the two sets?